### PR TITLE
Update language-server and testing libraries

### PR DESCRIPTION
**Pull request by msujew**
It seems like the `8.0.2` update of the language-server-protocol library contains a breaking change, see https://github.com/langium/langium/discussions/674. Users downloading that update will encounter a breaking change in Langium.

Also updates jest to `29.0.0`, that should also resolve the issue brought up in https://github.com/langium/langium/discussions/653.

### CODE REVIEW

File: packages/langium-vscode/package.json [hunk 1]
```diff
@@ -64,8 +64,7 @@
   },
   "dependencies": {
     "langium": "0.4.0",
-    "vscode-languageclient": "^8.0.1",
-    "vscode-languageserver": "^8.0.1",
+    "vscode-languageserver": "^8.0.2",
```
Comment (spoenemann) [inline packages/langium-vscode/package.json:67]: Why did you remove the client lib here? We use it in extension.ts

**Review by spoenemann — APPROVED** [review]

**Comment by spoenemann** [comment]
Test failure:

> packages/langium-cli/test/generator/module-generator.test.ts:21:34 - error TS2339: Property 'not' does not exist on type 'Matchers<string, {}>'.

File: packages/langium-vscode/package.json (hunk 1 above)
Comment (spoenemann) [reply packages/langium-vscode/package.json:67]: @msujew what about this? Seems to work anyway because of hoisted dependencies, but it seems unclean.

Comment (msujew) [reply packages/langium-vscode/package.json:67]: Ups, forgot about this. I'll create another PR.
